### PR TITLE
Bats test for Ockam examples

### PR DESCRIPTION
**Pull request by metaclips**
## Checks

- [ ] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [ ] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [ ] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam/edit/develop/.github/CONTRIBUTORS.csv) file in the github web UI and create a separate Pull Request, this will mark the commit as verified.

### CODE REVIEW

File: examples/command/portals/kafka/aiven/application_team/docker-compose.yml [hunk 1]
```diff
@@ -30,10 +32,13 @@ services:
           echo "Press Ctrl+C to stop the container at any time, then execute "\
           "'./run.sh cleanup' to remove all containers.";
           echo "";
-          kafka-console-consumer.sh --topic demo --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8;
-          if [ "$?" == "0" ]; then
-            echo "The example run was successful 🥳";
+          kafka-console-consumer.sh --topic demo --timeout-ms 50000 --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8 | tee output.txt;
+          if [[ ! -s output.txt ]]; then
+            echo "Failed to received messages";
```
Comment (etorreborre) [inline examples/command/portals/kafka/aiven/application_team/docker-compose.yml]: s/received/receive

File: examples/command/portals/kafka/apache/docker/application_team/docker-compose.yml [hunk 2]
```diff
@@ -25,17 +28,20 @@ services:
           echo "Press Ctrl+C to stop the container at any time, then execute "\
           "'./run.sh cleanup' to remove all containers.";
           echo "";
-          kafka-console-consumer.sh --topic demo --bootstrap-server 127.0.0.1:9092 --max-messages 8;
-          if [ "$?" == "0" ]; then
-            echo "The example run was successful 🥳";
+          kafka-console-consumer.sh --topic demo --timeout-ms 50000 --bootstrap-server 127.0.0.1:9092 --max-messages 8 | tee output.txt;
+          if [[ ! -s output.txt ]]; then
+            echo "Failed to received messages";
```
Comment (etorreborre) [inline examples/command/portals/kafka/apache/docker/application_team/docker-compose.yml]: s/received/receive

File: examples/command/portals/kafka/confluent/application_team/docker-compose.yml [hunk 3]
```diff
@@ -29,10 +31,13 @@ services:
           echo "Press Ctrl+C to stop the container at any time, then execute "\
           "'./run.sh cleanup' to remove all containers.";
           echo "";
-          kafka-console-consumer.sh --topic demo --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8;
-          if [ "$?" == "0" ]; then
-            echo "The example run was successful 🥳";
+          kafka-console-consumer.sh --topic demo --timeout-ms 50000 --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8 | tee output.txt;
+          if [[ ! -s output.txt ]]; then
+            echo "Failed to received messages";
```
Comment (etorreborre) [inline examples/command/portals/kafka/confluent/application_team/docker-compose.yml]: s/received/receive

File: examples/command/portals/kafka/instaclustr/docker/application_team/docker-compose.yml [hunk 4]
```diff
@@ -28,12 +30,14 @@ services:
...
           echo "Press Ctrl+C to stop the container at any time, then execute './run.sh cleanup' to remove all containers.";
           echo "";
-          kafka-console-consumer.sh --topic demo --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8;
-          if [ "$?" -eq 0 ]; then
-            echo "Messages received successfully";
-            echo "The example run was successful 🥳";
-            echo "The consumer was able to decrypt messages, Instaclustr only saw encrypted messages";
+          kafka-console-consumer.sh --topic demo --timeout-ms 50000 --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8 | tee output.txt;
+          if [[ ! -s output.txt ]]; then
+            echo "Failed to received messages";
```
Comment (etorreborre) [inline examples/command/portals/kafka/instaclustr/docker/application_team/docker-compose.yml]: s/received/receive

File: examples/command/portals/kafka/redpanda/docker/application_team/docker-compose.yml [hunk 5]
```diff
@@ -28,19 +31,24 @@ services:
...
           echo "";
-          kafka-console-consumer.sh --topic demo --bootstrap-server 127.0.0.1:9092 --max-messages 8;
-          if [ "$?" == "0" ]; then
-            echo "";
-            echo "The example run was successful 🥳";
-            echo "The consumer was able to decrypt messages, Redpanda only saw encrypted messages";
-            echo "";
+          kafka-console-consumer.sh --topic demo --timeout-ms 50000 --bootstrap-server 127.0.0.1:9092 --max-messages 8 | tee output.txt;
+          if [[ ! -s output.txt ]]; then
+            echo "Failed to received messages";
```
Comment (etorreborre) [inline examples/command/portals/kafka/redpanda/docker/application_team/docker-compose.yml]: s/received/receive

File: examples/command/portals/kafka/warpstream/application_team/docker-compose.yml [hunk 6]
```diff
@@ -31,15 +34,19 @@ services:
           echo "Press Ctrl+C to stop the container at any time, then execute "\
           "'./run.sh cleanup' to remove all containers.";
           echo "";
-          kafka-console-consumer.sh --topic demo --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8;
-          if [ "$$?" == "0" ]; then
-            echo "The example run was successful 🥳";
+          kafka-console-consumer.sh --topic demo --timeout-ms 50000 --bootstrap-server 127.0.0.1:9092 --consumer.config kafka.config --max-messages 8 | tee output.txt;
+          if [[ ! -s output.txt ]]; then
+            echo "Failed to received messages";
```
Comment (etorreborre) [inline examples/command/portals/kafka/warpstream/application_team/docker-compose.yml]: s/received/receive

**Review by etorreborre — DISMISSED** [review]
Very impressive!

File: implementations/rust/ockam/ockam_command/tests/bats/examples/databases.bats [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  setup_home_dir
+  skip_if_orchestrator_tests_not_enabled
+  copy_enrolled_home_dir
+}
+
+teardown() {
+  if [[ -z $BATS_TEST_COMPLETED ]]; then
+    echo "Test failed $(pwd)" >&3
+    cat log.txt >&3 || true
+  fi
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/tests/bats/examples/databases.bats:19]: Isn't this similar to what we do at [`teardown_home_dir()`](https://github.com/build-trust/ockam/blob/2239381d4ddf4589594343bdb9068202bafb0a71/implementations/rust/ockam/ockam_command/tests/bats/load/base.bash#L64)? Should we modify that function instead?

**Review by adrianbenavides — DISMISSED** [review]
Nice!! The bats tests look really good 💪

File: implementations/rust/ockam/ockam_command/tests/bats/examples/databases.bats (hunk 7 above)
Comment (metaclips) [reply implementations/rust/ockam/ockam_command/tests/bats/examples/databases.bats:19]: I don't think that's needed, as we only create the log.txt file in the examples test.

**Review by adrianbenavides — APPROVED** [review]

**Pull request merged**